apiplugin/file/filehost.go

```diff
@@ -159,30 +168,35 @@ func (f *fileSvcImpl) open(ctx context.Context, req *file.OpenRequest,
 	fileDataCache := *f.fileDataCache
 
 	// if file doesn't exist, return an error
-	if fid, exist := fpathTofid[cleanPath]; !exist {
+	fid, exist := fpathTofid[cleanPath]
+	if !exist {
 		pcontext.Errorf(ctx, "file does not exist and cannot be opened: %s", fpath)
 
 		return int32(file.FileErr_NotExistError)
-	} else {
-		// check file status
-		if fileDataCache[fid].status == Fs_Open {
-			pcontext.Errorf(ctx, "file is open, cannot be opened again: %s", fpath)
+	}
 
-			return int32(file.FileErr_AlreadyInUseError)
-		}
+	myFileInfo := fileDataCache[fid]
+	// check file status
+	if myFileInfo.status == Fs_Open {
+		pcontext.Errorf(ctx, "file is open, cannot be opened again: %s", fpath)
 
-		resp.Id = fid.Marshal()
-		fileDataCache[fid].lastAccessTime = time.Now()
-		fileDataCache[fid].status = Fs_Open
+		return int32(file.FileErr_AlreadyInUseError)
 	}
 
+	resp.Id = fid.Marshal()
+
+	myFileInfo.lastAccessTime = pcontext.CurrentTimeString(ctx, true)
```

Comment:
You probably want this to be a time object, not a string.  You can always convert it to string (or the code that calls yours can convert it) when you are presenting it to a human.